simd/2022/SIMD-08-05.md

```diff
@@ -0,0 +1,193 @@
+
+JM: for performance numbers, have anything more holistic? not just instruction itself, but a more representative use case?
+
+MD: no complete matrix multiplication benchmark, expect to have it over the weekend
+
+JM: would like to know with the 2 instructions what performance will be. If there is a risk of compat difference between 2 versions of ARM chips, that's not something we want to risk.
+
+MD: 2 ways to lower, directly bf dot instructions in bf16 instruction, this has different semantics than simulation code (even, odd, fmas). ARM also has bfmla instructions, specifically for this use case, when we want to bit exact with emulation code with previous generation CPUs. These instructions come with cost, twice as expensive as bfdot, but are on today's CPU. if we target bit-exact compat, then it is acceptable.
+
+JM: want to make sure we have this data point, thanks
+
+DG: from chrome perspective, still trying to get user data of how we slice down user stats, for fingerprinting, depending on how many people are on the newer subset. We still have older users we have to support, don't want to increase fingerprinting. Engines can dynamically make that decision, depending on what exactly we want to support, how we see that performance to deterministic trade off. Risk of fingerprinting on this instruction seems high. Not opposed to include in spec, but won't see us getting the performance off the bat.
+
+MD: would be acceptable to start with 2 instructions today, bfloat16 in armv9, and eventually all CPUs will be on armv9 and later, then we can switch to bfdot
+
+DG: to JM, curious about how you think about including something in spec v.s. how deterministic engine can be
+
+JM: agree with you so far, Apple cares a lot about fingerprinting, in this instance more concerned about compatibility. Haven't seen a lot of applications using relaxed simd, my impression is that this instruction is very early, most users don't have a computer that supports it, from that perspective, we should be cautious to not introduce compatibility cliffs
+
+MD: is your concern about performance cliffs when instruction is simulated?
+
+JM: concerned about a bug where some website works fine in Intel but doesn't work on ARM. Acceptable if there is an enormous performance benefit. If it is 2%, seems not worth the risk or extra complexity. Other parts of the spec are clearer on performance benefit on wide range of data. Curious about actual performance of this instruction.
+
+MD: do you know if Apple M2 supports bfloat 16
+
+JM: no idea, haven't figured it out yet
+
+MD: Apple recently added M2 detection to cpuinfo library, probably already supported or maybe in future iterations
+
+JM: i don't have an M2 yet
+
+ZA: AI on Marat for 
+
+DG: where do we stand on adding new instructions to make it more future proof? we have to spin off relaxed-simd spec because we didn't include instructions, so we don't have to spin off another new proposals.
```

Comment:
Maybe start a section, something like "Meta discussion on adding instructions for future hardware" - just a suggestion, everything looks good

Reply:
Good suggestion, done.